airflow/providers/amazon/aws/hooks/emr.py

```diff
@@ -77,22 +81,93 @@ def get_cluster_id_by_name(self, emr_cluster_name: str, cluster_states: list[str
-        run_job_flow method.
-        Overrides for this config may be passed as the job_flow_overrides.
+        Create and start running a new cluster (job flow).
+
+        This method uses ``EmrHook.emr_conn_id`` to receive the initial Amazon EMR cluster configuration.
+        If ``EmrHook.emr_conn_id`` is empty or the connection does not exist, then an empty initial
+        configuration is used.
+
+        :param job_flow_overrides: Is used to overwrite the parameters in the initial Amazon EMR configuration
+            cluster. The resulting configuration will be used in the boto3 emr client run_job_flow method.
+
+        .. seealso::
+            - :ref:`Amazon Elastic MapReduce Connection <howto/connection:emr>`
+            - `API RunJobFlow <https://docs.aws.amazon.com/emr/latest/APIReference/API_RunJobFlow.html>`_
+            - `boto3 emr client run_job_flow method <https://boto3.amazonaws.com/v1/documentation/\
+               api/latest/reference/services/emr.html#EMR.Client.run_job_flow>`_.
         """
-        try:
-            emr_conn = self.get_connection(self.emr_conn_id)
-            config = emr_conn.extra_dejson.copy()
-        except AirflowNotFoundException:
-            config = {}
+        config = {}
+        if self.emr_conn_id:
+            try:
+                emr_conn = self.get_connection(self.emr_conn_id)
+            except AirflowNotFoundException:
+                warnings.warn(
+                    f"Unable to find Amazon Elastic MapReduce Connection ID {self.emr_conn_id!r}, "
+                    "using an empty initial configuration. If you want to get rid of this warning "
+                    "message please provide a valid `emr_conn_id` or set it to None.",
+                    UserWarning,
+                    stacklevel=2,
+                )
+            else:
+                if emr_conn.conn_type and emr_conn.conn_type != "emr":
+                    warnings.warn(
+                        "Amazon Elastic MapReduce Connection expected connection type 'emr', "
+                        f"Connection {self.emr_conn_id!r} has conn_type={emr_conn.conn_type!r}. "
+                        f"This connection might not work correctly.",
+                        UserWarning,
+                        stacklevel=2,
+                    )
+                config = emr_conn.extra_dejson.copy()
         config.update(job_flow_overrides)
 
         response = self.get_conn().run_job_flow(**config)
 
         return response
 
+    def test_connection(self):
+        """
+        Return failed state for test Amazon Elastic MapReduce Connection (untestable)
+
+        We need to overwrite this method because this hook is based on
+        :class:`~airflow.providers.amazon.aws.hooks.base_aws.AwsGenericHook`,
+        otherwise it will try to test connection to AWS STS by using the default boto3 credential strategy.
+        """
+        return False, f"{self.hook_name} Connection cannot be tested."
```

Comment:
This is definitely an outlier here, but IMO it would be nice to include _why_ the connection cannot be tested in the response from the API/UI. It cannot be tested because of a user error? configuration of the environment? something else? I could foresee some users being puzzled without additional context. WDYT?

Reply:
> why the connection cannot be tested in the response from the API/UI.

Oh actually by design of this connection. It more close to the Airflow Variable rather than Airflow Connection.

So we just store this key-value pairs for https://docs.aws.amazon.com/emr/latest/APIReference/API_RunJobFlow.html

So nothing to test actually. It is even hardly possible to validate which key are valid, since this information stored somewhere in `botocore` and depend on which version of AWS API it supported.

The main problem in logic of providers manager which is responsible for a decision is Connection testable or not.

https://github.com/apache/airflow/blob/c94f978a66a7cfc31b6d461bbcbfd0f2ddb2962e/airflow/providers_manager.py#L793-L800 

In our case `EmrHook` based on `AwsGenericHook` which has `test_connection` method so it pass connection to `emr_conn_id` which has no affect in `AwsGenericHook.test_connection`.

But I agree that message from the current PR might be not so clear for end-users.

Reply:
I change message but still not sure is better `'Amazon Elastic MapReduce' Airflow Connection cannot be tested by design.` ?

Reply:
What about: `'Amazon Elastic MapReduce' Airflow Connection cannot be tested, by design it stores only key/value pairs and does not make a connection to an external resource'`

Reply:
Let me check how it looks in the UI

Reply:
![image](https://user-images.githubusercontent.com/3998685/193118197-f7849c4d-0c9a-4a77-bbd1-e267e147e8c9.png)

Looks fine!